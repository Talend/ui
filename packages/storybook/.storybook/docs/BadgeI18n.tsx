import * as React from 'react';

import Badge from './Badge';

const I18nStatus = (props: React.FunctionComponent) => {
	return (
		<Badge {...props} icon="i18next">
			i18n
		</Badge>
	);
};

export default React.memo(I18nStatus);
