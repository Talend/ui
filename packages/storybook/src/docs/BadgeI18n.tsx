import { memo } from 'react';

import Badge, { BadgeProps } from './Badge';

type I18nStatusProps = Omit<BadgeProps, 'icon'>;
const I18nStatus = (props: I18nStatusProps) => {
	return (
		<Badge {...props} icon="i18next">
			i18n
		</Badge>
	);
};

export default memo(I18nStatus);
