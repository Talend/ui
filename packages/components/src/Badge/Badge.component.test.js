import React from 'react';
import { shallow, mount } from 'enzyme';

import BadgeI18n, { Badge } from './Badge.component';

describe('Badge', () => {
	it('should render by default', () => {
		// given
		const label = 'my label';
		// when
		const wrapper = mount(<Badge label={label} />);
		// then
		expect(wrapper.html()).toMatchSnapshot();
	});
	it('should render with i18n', () => {
		// given nothing
		// when
		const wrapper = shallow(<BadgeI18n />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
