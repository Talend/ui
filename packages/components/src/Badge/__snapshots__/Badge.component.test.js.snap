// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render by default 1`] = `
<div class="tc-badge theme-tc-badge tc-badge-display-large theme-tc-badge-display-large tc-badge-readonly theme-tc-badge-readonly">
  <div class="tc-badge-button theme-tc-badge-button">
    <div class="tc-badge-label theme-tc-badge-label">
      <div class="theme-tc-tooltip"
           aria-describedby="42"
      >
        <span class="tc-badge-label-text theme-tc-badge-label-text"
              aria-describedby="42"
        >
          my label
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge should render with i18n 1`] = `
<Badge
  disabled={false}
  i18n={
    Object {
      "changeLanguage": [Function],
      "getFixedT": [Function],
      "hasResourceBundle": [Function],
      "init": [Function],
      "isInitialized": true,
      "isMock": true,
      "language": "en",
      "languages": Array [
        "en",
      ],
      "loadNamespaces": [Function],
      "off": [Function],
      "on": [Function],
      "options": Object {},
      "reportNamespaces": ReportNamespaces {
        "usedNamespaces": Object {
          "tui-components": true,
        },
      },
      "services": Object {
        "backendConnector": Object {},
        "resourceStore": Object {
          "data": Object {},
        },
      },
      "store": Object {
        "data": Object {},
        "off": [Function],
        "on": [Function],
      },
      "t": [Function],
    }
  }
  selected={false}
  t={[Function]}
  tReady={true}
/>
`;
